### ISSUE TITLE
Reconcile `dir_exists()` and `is_directory()`

### ISSUE THREAD

**Issue by dscho** [gitgitgadget/git#230]
In https://github.com/git/git/blob/v2.22.0-rc1/builtin/clone.c#L885-L889, there is a function with the short and sweet name `dir_exists()` (which is the obvious companion to `file_exists()`). However, it is marked as `static`, so nobody can use it outside `builtin/clone.c`.

There is also [`is_directory()`](https://github.com/git/git/blob/v2.22.0-rc1/abspath.c#L3-L12) which obviously tries to do the very same, but it uses a name that few developers will think of when they see `file_exists()` and look for the equivalent function to see whether a given directory exists.

Let's reconcile these, probably by renaming `is_directory()` to `dir_exists()` and using it also in `builtin/clone.c`.

**Comment by r1walz** [comment]
> Let's reconcile these, probably by renaming is_directory() to dir_exists() and using it also in builtin/clone.c.

Maybe moving it to [`dir.c`](https://github.com/git/git/blob/v2.22.0-rc1/dir.c#L2318) and renaming in `clone.c`?

**Comment by dscho** [comment]
> Maybe moving it to [`dir.c`](https://github.com/git/git/blob/v2.22.0-rc1/dir.c#L2312) and renaming in `builtin/clone.c`?

I don't think so, as the current `dir_exists()` does not verify that the path refers to a directory. In essence, it actually does more of a `path_exists()`.
